src/api/base.ts

```diff
@@ -68,58 +32,127 @@ const apiAxiosInstance = axios.create({
+    .catch((err: unknown) => {
+      if (axios.isAxiosError(err)) {
+        throw new HttpError(
+          err.message,
+          err.response?.status,
+          err.response?.data,
+          axios.isCancel(err) ? HTTP_CANCELLED_ERROR : HTTP_ERROR
+        )
       }
-    }, 200)
-
-    return apiAxiosInstance({ ...config, ...opts })
-      .then((success) => success.data)
-      .finally(() => {
-        if (!opts.skipLoader) useAppSpinner.hide()
-        window.clearTimeout(wait)
-      })
-  },
+
+      if (err instanceof Error) {
+        throw new HttpError(err.message, 0)
+      }
+
+      throw new HttpError("An unknown error has occurred.", 0)
+    })
+    .finally(() => {
+      if (options.skipLoader !== true) {
+        useAppSpinner.hide()
+      }
+
+      window.clearTimeout(wait)
+    })
+}
+
+const BaseAPI: HttpClient = {
   get<T = any>(
     path: string,
-    opts: RequestOptions,
+    opts?: RequestOptions,
     params?: Record<string, unknown>
   ) {
-    return this.makeRequest<T>({ url: path, method: "GET", params }, opts)
+    return httpRequest<T>({ url: path, method: "GET", params }, opts || {})
+  },
+  delete<T = any>(path: string, opts?: RequestOptions) {
+    return httpRequest<T>({ url: path, method: "DELETE" }, opts || {})
   },
-  delete<T = any>(path: string, opts: RequestOptions) {
-    return this.makeRequest<T>({ url: path, method: "DELETE" }, opts)
+  patch<T = any>(
+    path: string,
+    data: Record<string, unknown> | FormData,
+    opts?: RequestOptions
+  ) {
+    return httpRequest<T>({ url: path, data, method: "PATCH" }, opts || {})
   },
   post<T = any>(
     path: string,
     data: Record<string, unknown> | FormData,
-    opts: RequestOptions
+    opts?: RequestOptions
   ) {
-    return this.makeRequest<T>({ url: path, data, method: "POST" }, opts)
+    return httpRequest<T>({ url: path, data, method: "POST" }, opts || {})
   },
   put<T = any>(
     path: string,
     data: Record<string, unknown> | FormData,
-    opts: RequestOptions
+    opts?: RequestOptions
   ) {
-    return this.makeRequest<T>(
+    return httpRequest<T>(
       {
         url: path,
         data,
         method: "PUT",
       },
-      opts
+      opts || {}
     )
   },
 }
 
 export default BaseAPI
+
+/**
+ * A convenience method for checking if a variable in a failed request has an http status code.
+ * This is most useful for checking for specific http status codes in error callbacks.
+ * @param err unknown
+ * @param codes number | number[]
+ * @returns boolean
+ */
+export const hasHttpErrStatus = (err: unknown, codes: number | number[]) => {
+  if (typeof codes === "number") {
+    codes = [codes]
+  }
+
+  if (isHttpError(err)) {
+    return codes.includes(err.status)
+  }
+
+  return false
+}
+
+/**
+ * A type guard for checking if a variable is in the shape of a HttpError
+ * @param err unknown
+ * @returns err is HttpError
+ */
+export const isHttpError = <T>(err: unknown): err is HttpError<T> => {
```

Comment:
Note the return annotation is `err is HttpError`.  This is your user defined type guard.  It's like using `typeof err === "string"`, but for user defined interfaces.  If the condition is truthy, then the variable inside that block will be inferred to be of type `HttpError`.